wolox-books/src/app/screens/sign-up/sign-up.component.ts

```diff
@@ -0,0 +1,58 @@
+import { Component, OnInit } from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+
+@Component({
+  selector: 'sign-up',
+  templateUrl: './sign-up.component.html',
+  styleUrls: ['./sign-up.component.scss']
+})
+export class SignUpComponent implements OnInit {
+
+  rForm: FormGroup;
+
+  constructor(private fb:FormBuilder) { 
+
+    this.rForm = fb.group({
+      'firstName': [],
+      'lastName': [],
+      'email': [],
+      'password': []
+    })
+  }
+
+  ngOnInit() {
+  }
+
+  signUp() {
+    let form = new Form(this.rForm.controls.firstName.value, this.rForm.controls.lastName.value, this.rForm.controls.email.value, this.rForm.controls.password.value);
+    console.log(form);
+  }
+
+}
+
+class Form {
+  user: User;
+
+
+  constructor(firstName, lastName, email, password) {
+    this.user = new User(firstName, lastName, email, password);
+  }
+}
+
+class User {
```

Comment:
this is a good idea but it shouldn't be placed here. it's a `model` that goes in a separate file under the `models` folder #